views/networking/firewall/index.erb

```diff
@@ -0,0 +1,62 @@
+<% @page_title = "Networking" %>
+<div>
+  <%== render("networking/tabbar") %>
+  <div id="firewalls" class="grid gap-6 mt-6">
+    <% if @firewalls.count > 0 %>
+    <div class="grid gap-6">
+      <div class="overflow-hidden rounded-lg shadow ring-1 ring-black ring-opacity-5 bg-white divide-y divide-gray-200">
+        <table class="min-w-full divide-y divide-gray-300">
+          <thead class="bg-gray-50">
+            <tr>
+              <th scope="col" class="py-3.5 pl-4 pr-3 text-left text-sm
+              font-semibold text-gray-900 sm:pl-6">Name</th>
+              <th scope="col" class="py-3.5 pl-4 pr-3 text-left text-sm
+              font-semibold text-gray-900 sm:pl-6">Description</th>
+            </tr>
+          </thead>
+          <tbody class="divide-y divide-gray-200 bg-white">
+            <% @firewalls.each do |fw| %>
+              <tr>
+                <td class="whitespace-nowrap py-4 pl-4 pr-3 text-sm font-medium
+                text-gray-900 sm:pl-6" scope="row">
+                  <% if Authorization.has_permission?(@current_user.id, "Firewall:view", fw[:id]) %>
+                    <a href="<%= @project_data[:path] %><%= fw[:path] %>"
+                    class="text-orange-600 hover:text-orange-700"><%= fw[:name]
+                    %></a>
+                  <% else %>
+                    <%= fw[:name] %>
+                  <% end %>
+                </td>
+                <td class="whitespace-nowrap py-4 pl-4 pr-3 text-sm font-medium text-gray-900 sm:pl-6" scope="row"><%= fw[:description] %></td>
+              </tr>
+            <% end %>
+          </tbody>
+        </table>
+      </div>
+    </div>
+    <div class="space-y-1">
+      <%== render(
+        "components/page_header",
+        locals: {
+          title: "",
+          right_items: has_project_permission("Firewall:create") ? [
+            render("components/button", locals: { text: "Create Firewall", link: "firewall/create" })
+          ] : []
+        }
+      ) %>
+    </div>
```

Comment:
We shouldn't use `components/page_header` component like that. You can use `components/button` component with right alignment.

Reply:
Updated